### DISCUSSION TITLE
How to add GDPR cookie consent button?

### DISCUSSION THREAD

**Discussion by luiscantero** [datenstrom/community#680] [Ask a question] [answered]
Is there an extension to easily add a General Data Protection Regulation (GDPR) cookie consent button?
When using a cookie for traffic analysis, a consent button is needed. If the user rejects the privacy policy, either no cookies are used or the user if forwarded to another website.
How are people solving this problem? Thanks.

**Comment by GiovanniSalmeri** [comment]
I have given a quick glance at the commonest solutions, and most of them require just the insertion of a JS code in each page. This can be easily accomplished by editing header.html. This library seems to me one of the easiest to configure and use (nothig to handwrite, the site has a configurator):

https://cookie-bar.eu/

It would be trivial to make a Yellow extension which uses this code, but even without an extension its use in a Yellow site requires no more than adding a line to header.html. Other vanilla JS libraries:

https://www.osano.com/cookieconsent (it seems the most used)
https://www.cookieconsent.com/
https://github.com/manucaralmo/GlowCookies
https://github.com/orestbida/cookieconsent
https://alex-d.github.io/Cookies-EU-banner/
https://www.codexworld.com/cookie-consent-popup-with-javascript/ (minimal, you can only accept)
https://codeshack.io/eu-cookie-consent-popup-javascript/ (minimal, you can only accept)

I do not know which of them is really GDPR-compliant, but I am sure that the simplest and most lightweight solution for the GDPR rules and intricacies is to fully avoid non-essential cookies 😄

**Comment by GiovanniSalmeri** [comment, marked as answer]
Well, I have written this little extension which adds web analytics *and* creates a cookie consent banner:

https://github.com/GiovanniSalmeri/yellow-analytics

All the work is performed by [GlowCookies](https://github.com/manucaralmo/GlowCookies), I have only adapted it to Yellow. There are still some rough edges, but the essential seems to work.

**Reply by luiscantero** [reply]
Thanks a lot, great work!

**Reply by GiovanniSalmeri** [reply]
You are welcome! I have now renamed the extension to https://github.com/GiovanniSalmeri/yellow-analytics for clarity.
